app/controllers/admin/merchants/items_controller.rb

```diff
@@ -0,0 +1,82 @@
+class Admin::Merchants::ItemsController < Admin::BaseController
+  def index
+    @merchant = Merchant.find(params[:merchant_id])
+    @items = @merchant.items
+    render 'items/index'
+  end
+
+  def new
+    @merchant = Merchant.find(params[:merchant_id])
+    @item = Item.new
+    render 'merchant/items/new'
+  end
+
+  def create
+    @merchant = Merchant.find(params[:merchant_id])
+    @item = @merchant.items.new(item_params)
+    check_default_image(@item)
+    if @item.save
+      flash[:alert] = 'New item saved successfully!'
+      redirect_to "/admin/merchants/#{@merchant.id}/items"
+    else
+      flash[:error] = @item.errors.full_messages.to_sentence
+      render 'merchant/items/new'
+    end
+  end
+
+  def edit
+    @item = Item.find(params[:id])
+    render 'merchant/items/edit'
+  end
+
+  def update
+    @item = Item.find(params[:id])
+    @merchant = Merchant.find(params[:merchant_id])
+    @item.update(item_params) if params[:item]
+    check_default_image(@item) unless params[:status]
+    if params[:status]
+      update_status
+    elsif @item.save
+      flash[:alert] = "Item #{@item.id} has been successfully updated!"
+      redirect_to "/admin/merchants/#{@merchant.id}/items"
+    else
+      flash[:error] = @item.errors.full_messages.to_sentence
+      render 'merchant/items/edit'
+    end
+  end
+
+  def destroy
+    item = Item.find(params[:id])
+    merchant = Merchant.find(params[:merchant_id])
+    Review.where(item_id: item.id).destroy_all
```

Comment:
I think this should be a model method in order to follow MVC convention.

Reply:
Good call, removed! This was given to us from the setup FYI. We should look out for more things like this!